user-experience/WFLY-18428_get_started_with_wildfly.adoc

```diff
@@ -0,0 +1,119 @@
+---
+categories:
+  - user-experience
+  - maven
+  - installation  
+---
+= WFLY-18428 - Improve the "Get Started" Experience with WildFly
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-18428[WFLY-18428] - Improve the "Get Started" Experience with WildFly
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-17651[WFLY-17651] - Add a getting started archetype
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:darran.lofthouse@redhat.com[Darran Lofthouse]
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly.org - source code for https://wildfly.org/
+* https://github.com/wildfly/wildfly-archetypes - Maven archetypes for WildFly
+
+=== Relevant Installation Types
+
+* Application installation with server provisioned with wildfly-maven-plugin
+
+== Requirements
+
+* It must be possible to create, build and run a WildFly application in less than 5 minutes (no counting download times and requirements setup)
+* The only requirement to get started with WildFly are:
+** having a working Java 11+ installation (verified with `java -v`)
+** having a working Maven 3.9+ installation (verified with `maven -v`)
+* The user must be able to create a new Maven project that provides the minimal configuration to provision WildFly and run integration tests.
+* The "get started" documentation must be up to date with the latest release of WildFly.
+
+=== Hard Requirements
+
+* WildFly provisioning and configuration is controlled by the `wildfly-maven-plugin`
+** We want to promote a modern way (sustainable, maintainable) to develop with WildFly. Any interaction with WildFly is controlled by the wildfly-maven-plugin. This implies that instructions to download WildFly server with zips or modifyings its XML configuration by hand are prohibited
+* Dependencies to Jakarta EE artifacts are controlled by the WildFly BOM for Jakarta EE
+* Integration tests are configured to use Arquillian against the WildFly server provisioned by the `wildfly-maven-plugin`
```

Comment:
+1, I'll update the doc to clarify the plan for adding guides on top of this get started. I also want to make sure that this proposal aligns with https://github.com/wildfly/wildfly-proposals/pull/448

Reply:
@fjuma @emmartins @darranl, I reformulated this proposal so that #448 handle the subsequent guides with the requirement that it must comply with this "get started" and be applied to it.

Reply:
hi @jmesnil , if I understood correctly you are proposing that here we focus only on getting the user started (with an app that we provide), since #448 will focus on app development?

Reply:
Thanks, Jeff! Just FYI, for the use-case driven guides, we'll want to look into modularization so that common pieces of user stories can be shared easily across guides, without needing to re-write the same content for multiple guides. As an example, there could be content that can be reused for both OpenShift related and k8s related guides, there could be content that can be reused for both WildFly and EAP, etc. 

@theashiot has been starting to look into how we could go about modularizing content.